interfaces/cython/cantera/yaml2ck.py

```diff
@@ -0,0 +1,1047 @@
+import math
+import re
+import argparse
+from pathlib import PurePath, Path
+from textwrap import fill
+import cantera as ct
+
+# number of calories in 1000 Joules
+CALORIES_CONSTANT = 4184.0
+
+# Conversion from 1 debye to coulomb-meters
+DEBYE_CONVERSION = 3.33564095e-30
+
+
+def Eformat(val, precision=7, exp_digits=3):
+    """
+    Formats numbers in a pretty format for printing
+
+    :param val:
+        Float to be formatted into a str
+    :param precision:
+        How many digits after the decimal
+    :param exp_digits:
+        How many digits the exponent has
+    """
+    if math.isinf(val) and val > 0:
+        return " inf"
+
+    elif math.isinf(val):
+        return "-inf"
+
+    # round to correct number of significant digits
+    if val != 0:
+        val = round(val, precision - int(math.floor(math.log10(abs(val)))) - 1)
+
+    s = f"{val: .{precision}e}"
+
+    mantissa, exponent = s.split("e")
+    exp_digits += 1  # +1 due to sign
+
+    return f"{mantissa}E{int(exponent):+0{exp_digits}d}"
+
+
+def build_header_text(solution, max_width=120):
+    """
+    Creates the header text for all output files
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    """
+    text = []
+
+    if "description" in solution.input_header:
+        text_wrap = lambda t: fill(
+            t,
+            width=max_width,
+            break_long_words=False,
+            break_on_hyphens=False,
+            initial_indent="! ",
+            subsequent_indent="!   ",
+        )
+
+        text = solution.input_header["description"].splitlines()
+        text = [text_wrap(line) for line in text]
+    else:
+        note = ""
+
+    ct_note = (
+        f"! Chemkin file was converted from a Cantera {ct.__version__} "\
+        "`solution` object"
+    )
+
+    has_ct_note = False
+    for n, line in enumerate(text):
+        if "! Chemkin file was converted from a Cantera" in line:
+            text[n] = ct_note
+            has_ct_note = True
+            break
+
+    if not has_ct_note:
+        text.append(f"!\n{ct_note}\n! \n\n")
+
+    return "\n".join(text)
+
+
+def build_species_text(solution, max_width=80, sort_elements=True, 
+                       sort_species=True):
+    """
+    Creates element and species declarations
+
+    :param solution:
+        The `solution` object being converted
+    :param max_width:
+        The maximum width of lines before they start to wrap
+    :param sort_elements:
+        A boolean flag to indicate if elements should be sorted
+    :param sort_species:
+        A boolean flag to indicate if species should be sorted
+    """
+
+    def build_elements(elements, atomic_weights):
+        if sort_elements:  # Sort elements by atomic weight
+            elements = [c for _, c in sorted(zip(atomic_weights, elements))]
+
+            if sort_elements:
+                elements = elements[::-1]
```

Comment:
Why is this here? If you're inside this condition, isn't `sort_elements` always true? I think sorting from lightest to heaviest makes the most sense, so I don't think you need this.

Reply:
When I realized that my tests were failing because they were not sorted the same depending on if it came through my code or from the test data set, I changed to sorting everything. This previously had an option to choose ascending or descending. It looks like I missed this when removing that option. 

I liked sorting from heaviest to lightest. I'm not a stickler on this though.